### DISCUSSION TITLE
Possible to extend provider/Custom resource

### DISCUSSION THREAD

**Discussion by aktowns** [pulumi/pulumi#7575] [Pulumi Providers] [answered]
Apologies if this is the wrong place to ask, I'm looking to create a AWS [resource](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ec2-subnet-network-acl-assoc.html) with an API that doesn't exist in pulumi or in the upstream terraform provider yet [[1]](https://github.com/hashicorp/terraform-provider-aws/pull/18807) [[2]](https://github.com/hashicorp/terraform-provider-aws/pull/1034) [[3]](https://github.com/hashicorp/terraform-provider-aws/issues/17648) was wondering if there was a way for now to use the AWS API and create a `CustomResource` that represents it, something like (F#)

```fsharp
type SubnetNetworkAclAssociationArgs() =
    inherit ResourceArgs()

    [<Input("vpcId", required = true)>]
    member val VpcId: Input<string> = null with get, set

    [<Input("subnetId", required = true)>]
    member val SubnetId: Input<string> = null with get, set

//[<AwsResourceType("aws:ec2/subnetNetworkAclAssociation:SubnetNetworkAclAssociation")>]
type SubnetNetworkAclAssociation(name: string, args: SubnetNetworkAclAssociationArgs, options) =
    inherit CustomResource("aws:ec2/subnetNetworkAclAssociation:SubnetNetworkAclAssociation", name, args, options)

    (* On Create? or some lifecycle hook? *)
        let client = AmazonEC2Client()
        // Get current associationId from subnet.. 
        let req = ReplaceNetworkAclAssociationRequest(AssociationId = "", NetworkAclId = "")
        client.ReplaceNetworkAclAssociationAsync(req)
    
    (* On Delete *) .. 

```

I'm assuming not :( but holding out a little hope :slightly_smiling_face:

**Comment by nimbinatus** [comment, marked as answer]
Sorry for the delayed answer! For this specific resource, we've got it in the [AWS Native provider](https://www.pulumi.com/registry/packages/aws-native/api-docs/ec2/subnetnetworkaclassociation/), and you can have both native provider and classic provider calls that interact well in the same program. However, to your specific question about getting something that's not in the provider yet, the ideal way is creating a [dynamic provider](https://www.pulumi.com/docs/intro/concepts/resources/dynamic-providers/). Unfortunately, dynamic providers aren't available in C#/F# just yet; but if you're comfortable enough to write in TypeScript, JavaScript, or Python, you could wrap the API in a dynamic provider and get all of the state management goodness.

**Reply by Makeshift** [reply]
I'm aware that this is an old question, but I have a very similar use-case of wanting to do something not currently available in the AWS provider. I can create a dynamic provider, but I would like to inherit the credentials that the AWS provider is using to instantiate AWS-SDK. Is there a way of doing this?
